metamorph/plugins/morph_provision.py

```diff
@@ -0,0 +1,279 @@
+
+        :returns Dictionary of correct metadata by metadata location
+        """
+        for metadata_tree in metadata_source:
+            if metadata_location[0] in metadata_tree.keys():
+                return metadata_tree
+        raise ProvisionException('Unable to find key "{}" in given metadata file'.format(metadata_location[0]))
+
+    def update_topology_by_metadata(self, metadata_name, metadata_value):
+        """
+        Method for updating topology by metadata name and its value
+
+        :param metadata_name -- topology metadata name
+        :param metadata_value -- metadata name value
+        """
+        if metadata_name in self.res_defs.keys():
+            self.res_defs[metadata_name] = metadata_value
+            if metadata_name != 'networks':
+                self.res_defs[metadata_name] = str(self.res_defs[metadata_name])
+        elif metadata_name in self.resource_groups.keys():
+            self.resource_groups[metadata_name] = str(metadata_value)
+        elif metadata_name in self.provision_topology.keys():
+            self.provision_topology[metadata_name] = str(metadata_value)
+        else:
+            raise ProvisionException('Unable to find given metadata name "{}". '
+                                     'Name does not exists in topology file'.format(metadata_name))
+
+    def get_metadata_from_location(self, metadata_source, metadata_location, metadata_name):
+        """
+        Method for getting metadata value from metadata file by it's path
+
+        :param metadata_source -- extracted metadata from --metadata-file parameter
+        :param metadata_location -- searching metadata location
+        :param metadata_name -- name of metadata searched in given metadata location
+
+        :returns found metadata
+        """
+        if type(metadata_source) is list:
+            metadata_source = self.get_correct_metadata_tree(metadata_source, metadata_location)
+        if metadata_name in metadata_source.keys():
+            if type(metadata_source[metadata_name]) is dict:
+                raise ProvisionException("Metadata value can not be dictionary type. For more information see"
+                                         " documentation.")
+            return metadata_source[metadata_name]
+        elif not metadata_source:
+            raise ProvisionException('Unable to find metadata "{}" in given metadata location'.format(metadata_name))
+        return self.get_metadata_from_location(metadata_source[metadata_location[0]],
+                                               metadata_location[1:],
+                                               metadata_name)
+
+    @staticmethod
+    def get_git_repo_name(git_repo):
+        """
+        Method for getting git repository name
+
+        :param git_repo -- git repository path
+
+        :returns repository name
+        """
+        if git_repo.endswith('/'):
+            git_repo = git_repo[:-1]
```

Comment:
easier just to do:
```
git_repo = git_repo.rstrip('/')
```
If git_repo ends with '/' it'll strip it. If it doesn't end with '/' git_repo will be unaffected.